### PR TITLE
FOGL-2789: South service crash when adding python27 filter

### PR DESCRIPTION
**Pull request by pintomax**
FOGL-2789: South service crash when adding python27 filter

Centos/RHEL issue:
$ sudo yum install -y sudo yum install python27
$ cmake .. && make
$ foglamp start

Create a South Service ABC with Random C++ South plugin
Add filter PI65 with readings27.py script (in this repo)

Log messages

May 16 13:50:09 centos76 ABC[19642]: FATAL: Filter 'python27', script 'abc_pi65_script_readings27': Error '/opt/rh/python27/root/usr/lib64/python2.7/lib-dynload/_struct.so: undefined symbol: PyFloat_Type'

PATH vars from environment

LD_LIBRARY_PATH=/opt/rh/python27/root/usr/lib64:/opt/rh/rh-python36/root/usr/lib64:/usr/local/foglamp/lib
PATH=/opt/rh/python27/root/usr/bin:/opt/rh/rh-python36/root/usr/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/massimiliano/.local/bin:/home/massimiliano/bin
PKG_CONFIG_PATH=/opt/rh/python27/root/usr/lib64/pkgconfig:/opt/rh/rh-python36/root/usr/lib64/pkgconfig

### CODE REVIEW

**Review by MarkRiddoch — APPROVED** [review]
Looks ok to me. Not tried to run it however.

File: plugin.cpp [hunk 1]
```diff
@@ -126,24 +129,23 @@ PLUGIN_HANDLE plugin_init(ConfigCategory* config,
...
-	// Embedded Python 2.7 program name
-        Py_SetProgramName((char *)config->getName().c_str());
-	// Embedded Python 2.7 initialisation
-        Py_Initialize();
-
 	// Pass FogLAMP Data dir
 	pyFilter->setFiltersPath(getDataDir());
 
+	// Embedded Python 2.7 program name
+	Py_SetProgramName((char *)config->getName().c_str());
```
Comment (AmandeepArora) [inline plugin.cpp]: https://docs.python.org/2/c-api/init.html - Py_SetProgramName - This function should be called before Py_Initialize() is called for the first time, if it is called at all.

File: plugin.cpp [hunk 2]
```diff
@@ -153,14 +155,32 @@ PLUGIN_HANDLE plugin_init(ConfigCategory* config,
...
+		return (PLUGIN_HANDLE)info;
+	}
+
 	// Configure filter
 	if (!pyFilter->configure())
 	{
+		if (pythonInitialised)
+		{
+			pythonInitialised = false;
+			PyGILState_Release(state);
```
Comment (AmandeepArora) [inline plugin.cpp]: PyGILState_Release() should be called outside this if condition.

**Comment by pintomax** [comment]
@AmandeepArora  If changes are ok, we can can close the FOGL-2789 bug and create a nerw JIRA for CentOS issue

**Review by AmandeepArora — APPROVED** [review]

**Comment by AmandeepArora** [comment]
The centos issue seems to be due to 2 versions of python. See here - 

https://stackoverflow.com/questions/27383054/python-importerror-usr-local-lib-python2-7-lib-dynload-io-so-undefined-symb

**Pull request merged**